components/Search/index.js

```diff
@@ -0,0 +1,100 @@
+import algoliasearch from "algoliasearch/lite";
+import { useState, useEffect } from "react";
+import {
+  InstantSearch,
+  // SearchBox,
+  Hits,
+  connectSearchBox,
+} from "react-instantsearch-dom";
+import Topics from "@components/Topics";
+import Link from "next/link";
+import ReactMarkdown from "react-markdown";
+import ReactMarkdownRenderers from "@utils/ReactMarkdownRenderers";
+import PublishedDateAndReadingTime from "@components/Post/PublishedDateAndReadingTime";
+/**
+ * Styles are in main.styles.json
+ */
+
+const searchClient = algoliasearch(
+  process.env.NEXT_PUBLIC_ALGOLIA_APP_ID,
+  process.env.NEXT_PUBLIC_ALGOLIA_SEARCH_API_KEY,
+);
+
+const SearchBox = ({ currentRefinement, isSearchStalled, refine }) => (
+  <form noValidate action="" role="search">
+    <input
+      type="search"
+      value={currentRefinement}
+      onChange={(event) => refine(event.currentTarget.value)}
+    />
+    <button onClick={() => refine("")}>Reset query</button>
+    {isSearchStalled ? "My search is stalled" : ""}
+  </form>
+);
+
+const CustomSearchBox = connectSearchBox(SearchBox);
+
+const Hit = (props) => {
+  return (
+    <>
+      <PublishedDateAndReadingTime
+        date={props.hit.date}
+        readingTime={props.hit.readingTime}
+      />
+      <Link href={`/blog/${props.hit.slug}`}>
+        <a className="ais-Hits-link">{props.hit.title}</a>
+      </Link>
+      <Topics topics={props.hit.topics} />
+      <div className="ais-Hits-excerpt">
+        <ReactMarkdown
+          children={props.hit.excerpt}
+          renderers={ReactMarkdownRenderers(props.hit.excerpt)}
+        />
+      </div>
+    </>
+  );
+};
+
+export default function Search() {
+  const [searchTerm, setSearchTerm] = useState("");
+
+  useEffect(() => {
+    /**
+     * To add a form label for a11y appropriately
+     */
+    const searchInput = document.querySelector(".ais-SearchBox-input");
+    searchInput.id = "algolia_search";
+  });
+
+  function searchBoxOnChange(value) {
+    setSearchTerm(value);
+  }
+
+  return (
+    <>
+      {searchClient && (
+        <div className="ais-SearchHolder">
+          <label className="ais-SearchLabel" htmlFor="algolia_search">
+            Search articles
+          </label>
+          <InstantSearch searchClient={searchClient} indexName="p4nth3rblog">
+            <CustomSearchBox />
+            {/* <SearchBox
+              onChange={(e) => searchBoxOnChange(e.target.value)}
```

Comment:
How can I use this onChange with the custom search box? The reason why I presumed the hits weren't working is because I only want to show the hits when the searchLength in my state was length > 3.

Not sure how to access the current value of the custom search box here unless I use some pure JS?

Reply:
I think I tried passing those functions into the custom search box, but the onChange needs to fire the refine() function I think?

Reply:
No, that prevents typing if the string is fewer than the allowed characters!

Reply:
Also, just tried on my sandbox, you'll need to get rid of the `value={currentRefinement}` https://codesandbox.io/s/epic-glade-drb8h?file=/src/App.js

Reply:
I feel like I need a callback to my parent component really.

Reply:
That allows me to type now, but... is there a way to hide all hits by default until there's a refinement from outside the component?

Reply:
You could do that by using a customHits connector I guess! And implement your logic within https://www.algolia.com/doc/api-reference/widgets/hits/react/#full-example

Reply:
ah ha, ok.. let's see...

Reply:
My Hits doesn't know the currentRefinement length though!

Reply:
I believe that if you use the connector for the hits, you will be able to access it directly by declaring it as a spreaded prop, like this: 
```javascript
const Hits = ({ currentRefinement, hits }) => (
```

That being said, I'm not an instantsearch expert :D